features/facebookImport/test/importers/ad-view-language-importer.test.js

```diff
@@ -14,140 +14,94 @@ const datasets = [
-        datasets.map(async ({ language, dataset }) => {
-            const zipFile = new ZipFileMock();
-            zipFile.addJsonEntry(RECENTLY_VIEWED_FILE_PATH, dataset);
-            const importingResult = await runRecentlyViewedAdsImporter(zipFile);
-
-            return { language, importingResult };
-        })
-    ).then((importingResults) => {
-        testInputs = importingResults.map(({ language, importingResult }) => {
-            const result = importingResult.result;
-            const facebookAccount = importingResult.facebookAccount;
-            const relatedAccounts = facebookAccount.relatedAccounts;
-            const [firstRelatedAccount, secondRelatedAccount] =
-                relatedAccounts.items;
-            return [
-                language,
-                [
-                    result,
-                    facebookAccount,
-                    relatedAccounts,
-                    firstRelatedAccount,
-                    secondRelatedAccount,
-                ],
-            ];
+datasets.forEach(([language, dataset]) =>
+    describe(`Import ad views from ${language} dataset`, () => {
+        let result = null;
+        let relatedAccounts = null;
+        let firstRelatedAccount = null;
+        let secondRelatedAccount = null;
+
+        beforeAll(async () => {
+            const importingResult = await runAdsImportForDataset(dataset);
+            result = importingResult.result;
+            relatedAccounts = importingResult.relatedAccounts;
+            [firstRelatedAccount, secondRelatedAccount] = relatedAccounts.items;
         });
-    });
-});
-
-describe("Import ad views", () => {
-    console.log(testInputs);
-    test.each(testInputs)(
-        "returns success status for %s dataset.",
-        (language, result) => {
-            expectImportSuccess(result);
-        }
-    );
-
-    test.each(testInputs)(
-        "has two related accounts for %s dataset.",
-        (language, { relatedAccounts }) => {
-            expect(relatedAccounts.count).toBe(2);
-        }
-    );
-
-    test.each(testInputs)(
-        "has three ads for %s dataset.",
-        (language, { relatedAccounts }) => {
-            expect(relatedAccounts.adsCount).toBe(3);
-        }
-    );
-});
-*/
+
+        it("returns success status", () => expectImportSuccess(result));
+
+        it("has two related accounts", () =>
+            expect(relatedAccounts.count).toBe(2));
+
+        it("has three ads", () => expect(relatedAccounts.adsCount).toBe(3));
+
+        it("has four ad views", () =>
+            expect(relatedAccounts.adViewsCount).toBe(4));
+
+        it("has correct related account ids for first account", () => {
+            expect(firstRelatedAccount.rawId).toBeUndefined();
+            expect(firstRelatedAccount.urlId).toBe("companyx.com");
+        });
+
+        it("has correct related account ids for second account", () => {
+            expect(secondRelatedAccount.rawId).toBe("99888776543223");
```

Comment:
Again, literals here. Which match one found later. Possibly a constant, or extracting it from test data using some other method than the business logic, would help.

Reply:
In this case we'd need indeed more logic to extract the relevant values.